modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateProcessor.java

```diff
@@ -19,18 +21,32 @@
 import org.elasticsearch.script.ScriptService;
 import org.elasticsearch.script.TemplateScript;
 
+import java.lang.ref.SoftReference;
 import java.time.ZoneId;
 import java.time.ZoneOffset;
 import java.time.ZonedDateTime;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.concurrent.ConcurrentMap;
 import java.util.function.Function;
+import java.util.function.Supplier;
 
 public final class DateProcessor extends AbstractProcessor {
 
     public static final String TYPE = "date";
+    private static final String CACHE_CAPACITY_SETTING = "es.ingest.date_processor.cache_capacity";
+    private static final Cache CACHE;
+
+    static {
+        var cacheSizeStr = System.getProperty(CACHE_CAPACITY_SETTING, "256");
```

Comment:
What's the reason for doing this as a system property instead of a Settings thing (and related, are we documenting this setting)?

Reply:
We don't expect users will have to change this value unless they see a big performance effect on their pipelines. In the opposite case, we wanted to provide an easy way to change the value if we receive an SDH (with the obvious memory consumption trade-off). I inspired by [this other PR#80902](https://github.com/elastic/elasticsearch/pull/80902/commits/76d7a445fdfaa1aa99b58176a856d2a48bc12804) by @joegallo

Reply:
+1, `git grep 'System.getProperty("es' | grep -vi test/` shows a good number of examples of this as the pattern for non-customer-facing internal knobs that we don't expect to need to twiddle (but that we're making possible to twiddle as a favor to our future selves).

Reply:
OK, just wanted to make sure it was intentional.